File: Codewars/Python/Removing_consequetive_strings_char.py
def dup(arry):
    for i in range(len(arry)):
        j = 1
        while j < len(arry[i]) - 1:
            if arry[i][j] == arry[i][j + 1] or arry[i][j] == arry[i][j - 1]:
                arry[i] = arry[i][:j] + arry[i][j + 1:]
                j -= 1
            j += 1
    
    return arry

File: Codewars/CPP/Removing_consequetive_strings_char.cpp
#include <vector>
#include <string>
#include <algorithm>

std::string say_no_to_consequtive_strings (std::string input) {
    for (int i = 1; i < input.length() - 1; i++)
    {
        if (input[i] == input[i + 1] || input[i] == input[i - 1])
        {
            input.erase(input.begin() + i);
            i--;
        }
    }
    
    return input;
}

std::vector<std::string> dup(std::vector<std::string> arr){
    std::for_each(arr.begin(), arr.end(), [](std::string& s) {s = say_no_to_consequtive_strings(s);});

    return arr;
}
